labs/b4.md

```diff
@@ -0,0 +1,167 @@
+
+## Debian: An Example
+
+In this class, we will be focused on using Debian. As noted before, Debian uses apt/dpkg as its package manager.
+
+To package manager is invoked by using apt. Before you do anything, it is good package to update the package list so your package manager can find and fetch the most up-to-date packages. To do that, run:
+
+`apt update`
+
+To find the package to install:
+
+`apt cache search [package|description]`
+
+To install a package, run:
+
+`apt install package`
+
+To remove a package, run:
+
+`apt remove package`
+
+Easy? Sometimes, you want to update the packages that you have installed, run:
+
+`apt dist-upgrade`
+
+There are also other commands, such as removing unneeded dependencies and purging packages but that is what the man pages are for.
+
+### Example
+
+We are going to install GCC for the next step of the lab.
+
+**This, and all other parts of the lab, should be done on your DigitalOcean VMs, as you cannot install packages on OCF machines.** 
+Please note that you can only connect to your VM from inside the Berkeley network, so you will have to either be on campus wifi, or you will have to SSH 
+through `ssh.ocf.berkeley.edu` (or a similar on-campus host) to access your VM. 
+You should be able to connect to it at `<OCF username>.decal.xcf.sh` by running `ssh <username>@<username>.decal.xcf.sh`.
+
+Simply run:
+
+`sudo apt install gcc`
+
+Now check if GCC is installed by running the following to check GCC's version:
+
+`gcc --v`
+
+Now install the package `ruby-dev` for the next parts of the lab.
+
+## But what about software that isn’t in the repositories/package lists?
+
+This is a very common and frustrating thing to occur. Just when you want to install a package, it appears that it is not listed in any of the repositories that are available. What can you do? There are three ways in going about this. 
+
+#### Warning:
+**Installing software not from a repository carries the same risks as installing software from a .exe or .msi on Windows.** Linux is popularly considered more secure than Windows or MacOS because of its use of repositories--but a malicious package installed manually can pwn your system as thoroughly as a malware executable can on other OSes. **Use common sense! Only manually install from sources and companies you trust.**
+
+*(For users of Ubuntu: PPAs, or third-party repositories, carry the same risks of running malicious third-party code.)*
+
+If the developers provide a package compatible with your distribution (.deb for Debian-based distros), you can download that package and install it using:
+
+`apt install ./package.deb`
+
+Another way is to find a generic binary package from the developer. This comes in the form of shady shell scripts, or a binary tarball in which you can just extract it and run it, or an appimage, which is a special type of executable that brings its own dependencies.
```

Comment:
if we're going to tell them what these things are, and describe them as shady, we should explain how to identify them